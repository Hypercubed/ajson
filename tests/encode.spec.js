"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ava_1 = require("ava");
var _1 = require("../dist/");
var asjon = new _1.AJSON()
    .addProcessor(_1.jsonPointer)
    .addProcessor(_1.recurseObjects)
    .addProcessor(_1.recurseArrays)
    .addProcessor(_1.recurseMap)
    .addProcessor(_1.recurseSet)
    .addProcessor(_1.specialNumbers)
    .addProcessor(_1.undefinedValue)
    .addProcessor(_1.regexpValue)
    .addProcessor(_1.dateValue)
    .addProcessor(_1.symbolValue);
ava_1.test('prim values', function (t) {
    t.deepEqual(asjon.encode(42), 42, 'number');
    t.deepEqual(asjon.encode(-42), -42, 'negative number');
    t.deepEqual(asjon.encode('woo!!!'), 'woo!!!', 'string');
    t.deepEqual(asjon.encode(true), true, 'boolean');
    t.deepEqual(asjon.encode(null), null, 'null');
});
ava_1.test('special values', function (t) {
    t.deepEqual(asjon.encode(-0), { $numberDecimal: '-0' }, 'negative zero');
    t.deepEqual(asjon.encode(undefined), { $undefined: true }, 'undefined');
    t.deepEqual(asjon.encode(NaN), { $numberDecimal: 'NaN' }, 'NaN');
    t.deepEqual(asjon.encode(Infinity), { $numberDecimal: 'Infinity' }, 'Infinity');
    t.deepEqual(asjon.encode(-Infinity), { $numberDecimal: '-Infinity' }, 'Neg Infinity');
    t.deepEqual(asjon.encode(/regexp/img), { $regex: 'regexp', $options: 'gim' }, ' RegExp');
    t.deepEqual(asjon.encode(new Date(1e12)), { $date: '2001-09-09T01:46:40.000Z' }, 'Date');
    t.deepEqual(asjon.encode(Symbol('a')), { $symbol: 'a' }, 'Symbol');
});
ava_1.test('objects', function (t) {
    t.deepEqual(asjon.encode(['a', 'b', 'c']), ['a', 'b', 'c'], 'Array');
    t.deepEqual(asjon.encode([]), [], 'Array (empty)');
    t.deepEqual(asjon.encode({ foo: 'bar', 'x-y': 'z' }), { foo: 'bar', 'x-y': 'z' }, 'Object');
    t.deepEqual(asjon.encode(new Set([1, 2, 3])), { $set: [1, 2, 3] }, 'Set');
    t.deepEqual(asjon.encode(new Map([['a', 'b']])), { $map: [['a', 'b']] }, 'Map');
});
ava_1.test('objects with special values', function (t) {
    t.deepEqual(asjon.encode(['a',
        new Date(1e12), 'c']), ['a', { $date: '2001-09-09T01:46:40.000Z' }, 'c'], 'Array');
    t.deepEqual(asjon.encode({ foo: 'bar', 'x-y': new Date(1e12) }), { foo: 'bar', 'x-y': { $date: '2001-09-09T01:46:40.000Z' } }, 'Object');
    t.deepEqual(asjon.encode(new Set([1, new Date(1e12), 3])), { $set: [1, { $date: '2001-09-09T01:46:40.000Z' }, 3] }, 'Set');
    t.deepEqual(asjon.encode(new Map([['a', new Date(1e12)]])), { $map: [['a', { $date: '2001-09-09T01:46:40.000Z' }]] }, 'Map');
});
ava_1.test('self referential', function (t) {
    var self = {};
    self.a = self;
    t.deepEqual(asjon.encode(self), { a: { $ref: '#' } }, 'object');
    var arr = [];
    arr[0] = arr;
    t.deepEqual(asjon.encode(arr), [{ $ref: '#' }], 'array');
    var map = new Map();
    map.set('self', map);
    t.deepEqual(asjon.encode(map), { $map: [['self', { $ref: '#' }]] }, 'Map');
    var set = new Set();
    set.add(set);
    set.add(42);
    t.deepEqual(asjon.encode(set), { $set: [{ $ref: '#' }, 42] }, 'Set');
});
ava_1.test('deep', function (t) {
    var obj = {
        string: 'a_string',
        number: 42,
        decimal: 42.3,
        array: ['a_string', 42, 42.3],
        object: {
            'also a string': 'string',
            'a number': 42
        }
    };
    t.snapshot(asjon.encode(obj));
});
ava_1.test('all', function (t) {
    var obj = {
        _id: '5aa882d3638a0f580d92c677',
        index: 0,
        name: {
            first: 'Valenzuela',
            last: 'Valenzuela'
        },
        registered: new Date(2014, 0, 1),
        symbol: Symbol('banana'),
        range: [
            -Infinity,
            0,
            1,
            2,
            3,
            4,
            5,
            6,
            7,
            8,
            Infinity
        ],
        friends: [
            {
                id: -0,
                name: 'Benton Chase'
            },
            {
                id: 1,
                name: 'Mccarthy Morgan'
            },
            {
                id: NaN,
                name: 'Kaufman Price'
            }
        ]
    };
    obj.friends.push(obj);
    obj.friends.push(obj.friends[0]);
    t.snapshot(asjon.encode(obj));
});
ava_1.test('references', function (t) {
    var a = { a: 1, b: {} };
    a.c = a.b;
    t.deepEqual(asjon.encode(a), { a: 1, b: {}, c: { $ref: '#/b' } });
    var b = { a: 1, b: { c: {} } };
    b.c = b.b.c;
    t.deepEqual(asjon.encode(b), { a: 1, b: { c: {} }, c: { $ref: '#/b/c' } });
});
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiZW5jb2RlLnNwZWMuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyJlbmNvZGUuc3BlYy50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOztBQUFBLDJCQUEyQjtBQUMzQiw2QkFNa0I7QUFFbEIsSUFBTSxLQUFLLEdBQUcsSUFBSSxRQUFLLEVBQUU7S0FDdEIsWUFBWSxDQUFDLGNBQVcsQ0FBQztLQUN6QixZQUFZLENBQUMsaUJBQWMsQ0FBQztLQUM1QixZQUFZLENBQUMsZ0JBQWEsQ0FBQztLQUMzQixZQUFZLENBQUMsYUFBVSxDQUFDO0tBQ3hCLFlBQVksQ0FBQyxhQUFVLENBQUM7S0FDeEIsWUFBWSxDQUFDLGlCQUFjLENBQUM7S0FDNUIsWUFBWSxDQUFDLGlCQUFjLENBQUM7S0FDNUIsWUFBWSxDQUFDLGNBQVcsQ0FBQztLQUN6QixZQUFZLENBQUMsWUFBUyxDQUFDO0tBQ3ZCLFlBQVksQ0FBQyxjQUFXLENBQUMsQ0FBQztBQUU3QixVQUFJLENBQUMsYUFBYSxFQUFFLFVBQUEsQ0FBQztJQUNuQixDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsRUFBRSxDQUFDLEVBQUUsRUFBRSxFQUFFLFFBQVEsQ0FBQyxDQUFDO0lBQzVDLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsRUFBRSxFQUFFLGlCQUFpQixDQUFDLENBQUM7SUFDdkQsQ0FBQyxDQUFDLFNBQVMsQ0FBQyxLQUFLLENBQUMsTUFBTSxDQUFDLFFBQVEsQ0FBQyxFQUFFLFFBQVEsRUFBRSxRQUFRLENBQUMsQ0FBQztJQUN4RCxDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsSUFBSSxDQUFDLEVBQUUsSUFBSSxFQUFFLFNBQVMsQ0FBQyxDQUFDO0lBQ2pELENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxJQUFJLENBQUMsRUFBRSxJQUFJLEVBQUUsTUFBTSxDQUFDLENBQUM7QUFDaEQsQ0FBQyxDQUFDLENBQUM7QUFFSCxVQUFJLENBQUMsZ0JBQWdCLEVBQUUsVUFBQSxDQUFDO0lBQ3RCLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxDQUFDLENBQUMsQ0FBQyxFQUFFLEVBQUUsY0FBYyxFQUFFLElBQUksRUFBQyxFQUFFLGVBQWUsQ0FBQyxDQUFDO0lBQ3hFLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxTQUFTLENBQUMsRUFBRSxFQUFDLFVBQVUsRUFBRSxJQUFJLEVBQUMsRUFBRSxXQUFXLENBQUMsQ0FBQztJQUN0RSxDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsR0FBRyxDQUFDLEVBQUUsRUFBQyxjQUFjLEVBQUUsS0FBSyxFQUFDLEVBQUUsS0FBSyxDQUFDLENBQUM7SUFDL0QsQ0FBQyxDQUFDLFNBQVMsQ0FBQyxLQUFLLENBQUMsTUFBTSxDQUFDLFFBQVEsQ0FBQyxFQUFFLEVBQUMsY0FBYyxFQUFFLFVBQVUsRUFBQyxFQUFFLFVBQVUsQ0FBQyxDQUFDO0lBQzlFLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxDQUFDLFFBQVEsQ0FBQyxFQUFFLEVBQUMsY0FBYyxFQUFFLFdBQVcsRUFBQyxFQUFFLGNBQWMsQ0FBQyxDQUFDO0lBQ3BGLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxXQUFXLENBQUMsRUFBRSxFQUFDLE1BQU0sRUFBRSxRQUFRLEVBQUUsUUFBUSxFQUFFLEtBQUssRUFBQyxFQUFFLFNBQVMsQ0FBQyxDQUFDO0lBQ3ZGLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxJQUFJLElBQUksQ0FBQyxJQUFJLENBQUMsQ0FBQyxFQUFFLEVBQUMsS0FBSyxFQUFFLDBCQUEwQixFQUFDLEVBQUUsTUFBTSxDQUFDLENBQUM7SUFDdkYsQ0FBQyxDQUFDLFNBQVMsQ0FBQyxLQUFLLENBQUMsTUFBTSxDQUFDLE1BQU0sQ0FBQyxHQUFHLENBQUMsQ0FBQyxFQUFFLEVBQUMsT0FBTyxFQUFFLEdBQUcsRUFBQyxFQUFFLFFBQVEsQ0FBQyxDQUFDO0FBQ25FLENBQUMsQ0FBQyxDQUFDO0FBRUgsVUFBSSxDQUFDLFNBQVMsRUFBRSxVQUFBLENBQUM7SUFDZixDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsQ0FBQyxHQUFHLEVBQUUsR0FBRyxFQUFFLEdBQUcsQ0FBQyxDQUFDLEVBQUUsQ0FBQyxHQUFHLEVBQUUsR0FBRyxFQUFFLEdBQUcsQ0FBQyxFQUFFLE9BQU8sQ0FBQyxDQUFDO0lBQ3JFLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxFQUFFLENBQUMsRUFBRSxFQUFFLEVBQUUsZUFBZSxDQUFDLENBQUM7SUFDbkQsQ0FBQyxDQUFDLFNBQVMsQ0FBQyxLQUFLLENBQUMsTUFBTSxDQUFDLEVBQUMsR0FBRyxFQUFFLEtBQUssRUFBRSxLQUFLLEVBQUUsR0FBRyxFQUFDLENBQUMsRUFBRSxFQUFFLEdBQUcsRUFBRSxLQUFLLEVBQUUsS0FBSyxFQUFFLEdBQUcsRUFBQyxFQUFFLFFBQVEsQ0FBQyxDQUFDO0lBQ3pGLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxJQUFJLEdBQUcsQ0FBQyxDQUFDLENBQUMsRUFBRSxDQUFDLEVBQUUsQ0FBQyxDQUFDLENBQUMsQ0FBQyxFQUFFLEVBQUUsSUFBSSxFQUFFLENBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUMsRUFBQyxFQUFFLEtBQUssQ0FBQyxDQUFDO0lBQ3pFLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxJQUFJLEdBQUcsQ0FBQyxDQUFDLENBQUMsR0FBRyxFQUFFLEdBQUcsQ0FBQyxDQUFDLENBQUMsQ0FBQyxFQUFFLEVBQUUsSUFBSSxFQUFFLENBQUMsQ0FBQyxHQUFHLEVBQUUsR0FBRyxDQUFDLENBQUMsRUFBQyxFQUFFLEtBQUssQ0FBQyxDQUFDO0FBQ2pGLENBQUMsQ0FBQyxDQUFDO0FBRUgsVUFBSSxDQUFDLDZCQUE2QixFQUFFLFVBQUEsQ0FBQztJQUNuQyxDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsQ0FBQyxHQUFHO1FBQzNCLElBQUksSUFBSSxDQUFDLElBQUksQ0FBQyxFQUFFLEdBQUcsQ0FBQyxDQUFDLEVBQ3JCLENBQUMsR0FBRyxFQUFFLEVBQUUsS0FBSyxFQUFFLDBCQUEwQixFQUFDLEVBQUUsR0FBRyxDQUFDLEVBQUUsT0FBTyxDQUFDLENBQUM7SUFDN0QsQ0FBQyxDQUFDLFNBQVMsQ0FBQyxLQUFLLENBQUMsTUFBTSxDQUN0QixFQUFFLEdBQUcsRUFBRSxLQUFLLEVBQUUsS0FBSyxFQUFFLElBQUksSUFBSSxDQUFDLElBQUksQ0FBQyxFQUFDLENBQUMsRUFDckMsRUFBRSxHQUFHLEVBQUUsS0FBSyxFQUFFLEtBQUssRUFBRSxFQUFFLEtBQUssRUFBRSwwQkFBMEIsRUFBRSxFQUFDLEVBQUUsUUFBUSxDQUFDLENBQUM7SUFDekUsQ0FBQyxDQUFDLFNBQVMsQ0FBQyxLQUFLLENBQUMsTUFBTSxDQUN0QixJQUFJLEdBQUcsQ0FBQyxDQUFDLENBQUMsRUFBRSxJQUFJLElBQUksQ0FBQyxJQUFJLENBQUMsRUFBRSxDQUFDLENBQUMsQ0FBQyxDQUFDLEVBQ2hDLEVBQUUsSUFBSSxFQUFFLENBQUMsQ0FBQyxFQUFFLEVBQUUsS0FBSyxFQUFFLDBCQUEwQixFQUFFLEVBQUUsQ0FBQyxDQUFDLEVBQUMsRUFBRSxLQUFLLENBQUMsQ0FBQztJQUNqRSxDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQ3RCLElBQUksR0FBRyxDQUFDLENBQUMsQ0FBQyxHQUFHLEVBQUUsSUFBSSxJQUFJLENBQUMsSUFBSSxDQUFDLENBQUMsQ0FBQyxDQUFDLENBQUMsRUFDakMsRUFBRSxJQUFJLEVBQUUsQ0FBQyxDQUFDLEdBQUcsRUFBRSxFQUFFLEtBQUssRUFBRSwwQkFBMEIsRUFBRSxDQUFDLENBQUMsRUFBQyxFQUFFLEtBQUssQ0FBQyxDQUFDO0FBQ3BFLENBQUMsQ0FBQyxDQUFDO0FBRUgsVUFBSSxDQUFDLGtCQUFrQixFQUFFLFVBQUEsQ0FBQztJQUN4QixJQUFNLElBQUksR0FBUSxFQUFFLENBQUM7SUFDckIsSUFBSSxDQUFDLENBQUMsR0FBRyxJQUFJLENBQUM7SUFDZCxDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsSUFBSSxDQUFDLEVBQUUsRUFBRSxDQUFDLEVBQUUsRUFBRSxJQUFJLEVBQUUsR0FBRyxFQUFDLEVBQUUsRUFBRSxRQUFRLENBQUMsQ0FBQztJQUUvRCxJQUFNLEdBQUcsR0FBVSxFQUFFLENBQUM7SUFDdEIsR0FBRyxDQUFDLENBQUMsQ0FBQyxHQUFHLEdBQUcsQ0FBQztJQUNiLENBQUMsQ0FBQyxTQUFTLENBQUMsS0FBSyxDQUFDLE1BQU0sQ0FBQyxHQUFHLENBQUMsRUFBRSxDQUFDLEVBQUUsSUFBSSxFQUFFLEdBQUcsRUFBRSxDQUFDLEVBQUUsT0FBTyxDQUFDLENBQUM7SUFFekQsSUFBTSxHQUFHLEdBQUcsSUFBSSxHQUFHLEVBQUUsQ0FBQztJQUN0QixHQUFHLENBQUMsR0FBRyxDQUFDLE1BQU0sRUFBRSxHQUFHLENBQUMsQ0FBQztJQUNyQixDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsR0FBRyxDQUFDLEVBQUUsRUFBRSxJQUFJLEVBQUUsQ0FBQyxDQUFDLE1BQU0sRUFBRSxFQUFFLElBQUksRUFBRSxHQUFHLEVBQUMsQ0FBQyxDQUFDLEVBQUMsRUFBRSxLQUFLLENBQUMsQ0FBQztJQUV6RSxJQUFNLEdBQUcsR0FBRyxJQUFJLEdBQUcsRUFBRSxDQUFDO0lBQ3RCLEdBQUcsQ0FBQyxHQUFHLENBQUMsR0FBRyxDQUFDLENBQUM7SUFDYixHQUFHLENBQUMsR0FBRyxDQUFDLEVBQUUsQ0FBQyxDQUFDO0lBQ1osQ0FBQyxDQUFDLFNBQVMsQ0FBQyxLQUFLLENBQUMsTUFBTSxDQUFDLEdBQUcsQ0FBQyxFQUFFLEVBQUUsSUFBSSxFQUFFLENBQUMsRUFBRSxJQUFJLEVBQUUsR0FBRyxFQUFFLEVBQUUsRUFBRSxDQUFDLEVBQUMsRUFBRSxLQUFLLENBQUMsQ0FBQztBQUN0RSxDQUFDLENBQUMsQ0FBQztBQUVILFVBQUksQ0FBQyxNQUFNLEVBQUUsVUFBQSxDQUFDO0lBQ1osSUFBTSxHQUFHLEdBQUc7UUFDVixNQUFNLEVBQUUsVUFBVTtRQUNsQixNQUFNLEVBQUUsRUFBRTtRQUNWLE9BQU8sRUFBRSxJQUFJO1FBQ2IsS0FBSyxFQUFFLENBQUMsVUFBVSxFQUFFLEVBQUUsRUFBRSxJQUFJLENBQUM7UUFDN0IsTUFBTSxFQUFFO1lBQ04sZUFBZSxFQUFFLFFBQVE7WUFDekIsVUFBVSxFQUFFLEVBQUU7U0FDZjtLQUNGLENBQUM7SUFFRixDQUFDLENBQUMsUUFBUSxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsR0FBRyxDQUFDLENBQUMsQ0FBQztBQUNoQyxDQUFDLENBQUMsQ0FBQztBQUVILFVBQUksQ0FBQyxLQUFLLEVBQUUsVUFBQSxDQUFDO0lBQ1gsSUFBTSxHQUFHLEdBQUc7UUFDVixHQUFHLEVBQUUsMEJBQTBCO1FBQy9CLEtBQUssRUFBRSxDQUFDO1FBQ1IsSUFBSSxFQUFFO1lBQ0osS0FBSyxFQUFFLFlBQVk7WUFDbkIsSUFBSSxFQUFFLFlBQVk7U0FDbkI7UUFDRCxVQUFVLEVBQUUsSUFBSSxJQUFJLENBQUMsSUFBSSxFQUFFLENBQUMsRUFBRSxDQUFDLENBQUM7UUFDaEMsTUFBTSxFQUFFLE1BQU0sQ0FBQyxRQUFRLENBQUM7UUFDeEIsS0FBSyxFQUFFO1lBQ0wsQ0FBQyxRQUFRO1lBQ1QsQ0FBQztZQUNELENBQUM7WUFDRCxDQUFDO1lBQ0QsQ0FBQztZQUNELENBQUM7WUFDRCxDQUFDO1lBQ0QsQ0FBQztZQUNELENBQUM7WUFDRCxDQUFDO1lBQ0QsUUFBUTtTQUNUO1FBQ0QsT0FBTyxFQUFFO1lBQ1A7Z0JBQ0UsRUFBRSxFQUFFLENBQUMsQ0FBQztnQkFDTixJQUFJLEVBQUUsY0FBYzthQUNyQjtZQUNEO2dCQUNFLEVBQUUsRUFBRSxDQUFDO2dCQUNMLElBQUksRUFBRSxpQkFBaUI7YUFDeEI7WUFDRDtnQkFDRSxFQUFFLEVBQUUsR0FBRztnQkFDUCxJQUFJLEVBQUUsZUFBZTthQUN0QjtTQUNGO0tBQ0YsQ0FBQztJQUVGLEdBQUcsQ0FBQyxPQUFPLENBQUMsSUFBSSxDQUFNLEdBQUcsQ0FBQyxDQUFDO0lBQzNCLEdBQUcsQ0FBQyxPQUFPLENBQUMsSUFBSSxDQUFDLEdBQUcsQ0FBQyxPQUFPLENBQUMsQ0FBQyxDQUFDLENBQUMsQ0FBQztJQUVqQyxDQUFDLENBQUMsUUFBUSxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsR0FBRyxDQUFDLENBQUMsQ0FBQztBQUNoQyxDQUFDLENBQUMsQ0FBQztBQUVILFVBQUksQ0FBQyxZQUFZLEVBQUUsVUFBQSxDQUFDO0lBQ2xCLElBQU0sQ0FBQyxHQUFRLEVBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsRUFBRSxFQUFDLENBQUM7SUFDN0IsQ0FBQyxDQUFDLENBQUMsR0FBRyxDQUFDLENBQUMsQ0FBQyxDQUFDO0lBQ1YsQ0FBQyxDQUFDLFNBQVMsQ0FBQyxLQUFLLENBQUMsTUFBTSxDQUFDLENBQUMsQ0FBQyxFQUFFLEVBQUMsQ0FBQyxFQUFFLENBQUMsRUFBRSxDQUFDLEVBQUUsRUFBRSxFQUFFLENBQUMsRUFBRSxFQUFDLElBQUksRUFBRSxLQUFLLEVBQUMsRUFBQyxDQUFDLENBQUM7SUFFOUQsSUFBTSxDQUFDLEdBQVEsRUFBQyxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsRUFBRSxFQUFFLENBQUMsRUFBRSxFQUFFLEVBQUUsRUFBQyxDQUFDO0lBQ3BDLENBQUMsQ0FBQyxDQUFDLEdBQUcsQ0FBQyxDQUFDLENBQUMsQ0FBQyxDQUFDLENBQUM7SUFDWixDQUFDLENBQUMsU0FBUyxDQUFDLEtBQUssQ0FBQyxNQUFNLENBQUMsQ0FBQyxDQUFDLEVBQUUsRUFBRSxDQUFDLEVBQUUsQ0FBQyxFQUFFLENBQUMsRUFBRSxFQUFFLENBQUMsRUFBRSxFQUFFLEVBQUMsRUFBRSxDQUFDLEVBQUUsRUFBRSxJQUFJLEVBQUUsT0FBTyxFQUFDLEVBQUMsQ0FBQyxDQUFDO0FBQzFFLENBQUMsQ0FBQyxDQUFDIn0=