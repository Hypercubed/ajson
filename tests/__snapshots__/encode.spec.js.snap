// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all 1`] = `
Object {
  "_id": "5aa882d3638a0f580d92c677",
  "friends": Array [
    Object {
      "id": Object {
        "$numberDecimal": "-0",
      },
      "name": "Benton Chase",
    },
    Object {
      "id": 1,
      "name": "Mccarthy Morgan",
    },
    Object {
      "id": Object {
        "$numberDecimal": "NaN",
      },
      "name": "Kaufman Price",
    },
    Object {
      "$ref": "#",
    },
    Object {
      "$ref": "#/friends/[0]",
    },
  ],
  "index": 0,
  "name": Object {
    "first": "Valenzuela",
    "last": "Valenzuela",
  },
  "range": Array [
    Object {
      "$numberDecimal": "-Infinity",
    },
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    Object {
      "$numberDecimal": "Infinity",
    },
  ],
  "registered": Object {
    "$date": "2014-01-01T07:00:00.000Z",
  },
  "symbol": Object {
    "$symbol": "banana",
  },
}
`;

exports[`deep 1`] = `
Object {
  "array": Array [
    "a_string",
    42,
    42.3,
  ],
  "decimal": 42.3,
  "number": 42,
  "object": Object {
    "a number": 42,
    "also a string": "string",
  },
  "string": "a_string",
}
`;
